packages/vite/src/node/__tests__/config.spec.ts

```diff
@@ -359,3 +359,63 @@ describe('resolveConfig', () => {
     await resolveConfig({ root: './inc?ud#s', customLogger: logger }, 'build')
   })
 })
+
+test('ssr config compat', async () => {
+  const config = await resolveConfig(
+    {
+      resolve: {
+        conditions: ['client1'],
+      },
+      ssr: {
+        resolve: {
+          conditions: ['ssr1'],
+        },
+      },
+      plugins: [
+        {
+          name: 'test',
+          config() {
+            return {
+              environments: {
+                client: {
+                  resolve: {
+                    conditions: ['client2'],
+                  },
+                },
+                ssr: {
+                  resolve: {
+                    conditions: ['ssr2'],
+                  },
+                },
+              },
+            }
+          },
+        },
+      ],
+    },
+    'serve',
+  )
+  expect(config.resolve.conditions).toMatchInlineSnapshot(`
+    [
+      "client1",
+    ]
+  `)
+  expect(config.environments.client.resolve.conditions).toMatchInlineSnapshot(`
+    [
+      "client1",
+      "client2",
+    ]
+  `)
+  expect(config.ssr.resolve?.conditions).toMatchInlineSnapshot(`
+    [
+      "ssr1",
+      "ssr2",
+    ]
+  `)
+  expect(config.environments.ssr.resolve?.conditions).toMatchInlineSnapshot(`
+    [
+      "ssr1",
+      "ssr2",
+    ]
+  `)
```

Comment:
`enviornments.client.resolve` is already merged with root `rosolve`

https://github.com/vitejs/vite/blob/0aae18925d778d70afc947969316a574d963234b/packages/vite/src/node/config.ts#L1137-L1142

I'm not sure why the difference of `resolve.conditions` and `ssr.resolve.conditions`. Let me take a closer look.

Reply:
In https://github.com/vitejs/vite/pull/18857/commits/e686c7de344251bfdbd0e138462286c6c0893451, I also made `config.enviornments.client.resolve` merge back to `config.resolve`

```js
 expect(config.resolve.conditions).toMatchInlineSnapshot(`
    [
      "client1",
      "client2",
    ]
  `)
```